soweego/validator/checks.py

```diff
@@ -81,25 +91,24 @@ def dead_ids_cli(catalog, entity, deprecate, sandbox, dump_wikidata, dir_io):
     you can pass the '-d' flag to do so.
     """
     dead_ids_path = os.path.join(
-        dir_io, DEAD_IDS_FILENAME.format(catalog, entity)
+        dir_io, DEAD_IDS_FNAME.format(catalog=catalog, entity=entity)
+    )
+    wd_cache_path = os.path.join(
+        dir_io,
+        WD_CACHE_FNAME.format(
+            catalog=catalog, entity=entity, criterion='dead_ids'
+        ),
     )
-    wd_ids_path = os.path.join(dir_io, WD_IDS_FILENAME.format(catalog, entity))
 
     # Handle Wikidata cache
-    if os.path.isfile(wd_ids_path):
-        with open(wd_ids_path) as wdin:
-            wd_ids = _load_wd_cache(wdin)
+    if os.path.isfile(wd_cache_path):
+        with open(wd_cache_path, 'rb') as cin:
+            wd_cache = pickle.load(cin)
+        LOGGER.info("Loaded Wikidata cache from '%s'", cin.name)
         # Discard the second return value: Wikidata cache
-        dead, _ = dead_ids(catalog, entity, wd_cache=wd_ids)
+        dead, _ = dead_ids(catalog, entity, wd_cache=wd_cache)
     else:
-        dead, wd_ids = dead_ids(catalog, entity)
-
-    # Dump ids gathered from Wikidata
-    if dump_wikidata:
-        _dump_wd_cache(wd_ids, wd_ids_path)
-        LOGGER.info(
-            'Identifiers gathered from Wikidata dumped to %s', wd_ids_path
-        )
+        dead, wd_cache = dead_ids(catalog, entity)
```

Comment:
I'd just set wd_cache to None and call dead_ids once. this will make possible future refactorings easier:

```
if os.path.isfile(wd_cache_path):
    wd_cache = ....
else:
    wd_cache = None

dead, wd_cache = dead_ids(catalog, entity)
```

and update later code to save the new cache if needed. (also see my other comment on having an object to handle all this loading and saving of the cache)

Reply:
Suggestion added to #421